### PR TITLE
Add support for OpenAddresses geojson format

### PR DESCRIPTION
**Pull request by Joxit**
:wave: I did some awesome work for the Pelias project and would love for everyone to have a look at it and provide feedback.

---
#### Here's the reason for this change :rocket:

I wanted to retrieve fresh data from OpenAddresses and saw that they had completely changed their data format. The data is now available on [batch.openaddresses.io](https://batch.openaddresses.io/data) and they serve ndjson instead of csv. 

---
#### Here's what actually got changed :clap:

I've updated the importer to support both csv and geojson files at import time.

### CODE REVIEW

**Comment by orangejulius** [comment]
Aw yes! I was wondering when someone was going to mention this big change :)

We'll test this out ASAP but it looks like the perfect approach: convert the incoming geojson to essentially the CSV format. That should work as long as the new format doesn't add interesting new data that we can't handle with the old CSV format.

Thanks @Joxit ❤️

**Comment by Joxit** [comment]
You are welcome :smile: 

It was a big surprise when I saw the new data page UI and format :sweat_smile: 
I did not change the download part because it's a bit more complicated with the new API to get filtered data... :confused:

File: lib/isValidCsvRecord.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ function streetContainsExclusionaryWord(record) {
 
 function hasAllProperties(record) {
   return [ 'LON', 'LAT', 'NUMBER', 'STREET' ].every(function(prop) {
-    return record[ prop ] && record[ prop ].length > 0;
+    return record[ prop ] && (typeof record[ prop ] === 'number' || record[ prop ].length > 0);
```
Comment (missinglink) [inline lib/isValidCsvRecord.js:32]: Looks like a job for... 🦸‍♂️ lodash?

File: lib/parameters.js [hunk 2]
```diff
@@ -95,8 +95,8 @@ function getFullFileList(peliasConfig, args) {
   const files = _.get(peliasConfig.imports.openaddresses, 'files', []);
 
   if (_.isEmpty(files)) {
-    // no specific files listed, so return all .csv files
-    return glob.sync( args.dirPath + '/**/*.csv' );
+    // no specific files listed, so return all .csv and .geojson files
+    return glob.sync( args.dirPath + '/**/*.csv' ).concat(glob.sync( args.dirPath + '/**/*.geojson' ));
```
Comment (missinglink) [inline lib/parameters.js:99]: I didn't test it out, but many glob libraries support 'braces':
```
/**/*.{csv,geojson}
```

Not a big deal, but might save a little disk spinning 😄

File: lib/streams/recordStream.js [hunk 3]
```diff
@@ -65,14 +59,52 @@ function createRecordStream( filePath, dirPath ){
...
+  return stream
+    .pipe(split())
+    .pipe(through.obj((line, _enc, next) => {
+      let row;
+      try {
+        const geojson = JSON.parse(line);
+        row = {
+          NUMBER: _.get(geojson, 'properties.number'),
+          STREET: _.get(geojson, 'properties.street'),
+          LON: _.get(geojson, 'geometry.coordinates[0]'),
```
Comment (missinglink) [inline lib/streams/recordStream.js:79]: Would it be worth adding a quick check that the geometry type is `Point` here?

I'm not really sure what the reasons were for moving to `.geojson` but presumably they are so they can expand this format to do things like centerlines which would be `LineString` and potentially cause issues here? 🤷‍♂️

**Review by missinglink — APPROVED** [review]
Looks good to me. I added some minor comments.

File: lib/isValidCsvRecord.js (hunk 1 above)
Comment (Joxit) [reply lib/isValidCsvRecord.js]: sure! :sweat_smile:

File: lib/parameters.js (hunk 2 above)
Comment (Joxit) [reply lib/parameters.js]: yes, this work like a charm :smile:

**Comment by Joxit** [comment]
Updated, thanks for your review :wink: 

Time to sleep now :sleeping:

**Comment by missinglink** [comment]
Haha the perils of timezones, turn your notifications off and go to bed! ;)

**Comment by orangejulius** [comment]
Wow, @Joxit. Coding 24/7 I see

File: lib/streams/recordStream.js (hunk 3 above)
Comment (missinglink) [reply lib/streams/recordStream.js]: Screenshot 2021-02-24 at 15 14 15

🔮

**Pull request merged**